src/main/scala/fs2/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -246,15 +246,23 @@ private[kafka] final class KafkaConsumerActor[F[_], K, V](
 
   private[this] def fetch(
     partition: TopicPartition,
+    streamId: Int,
     deferred: Deferred[F, (Chunk[CommittableMessage[F, K, V]], FetchCompletedReason)]
   ): F[Unit] = {
     val assigned =
       withConsumer { consumer =>
         F.delay(consumer.assignment.contains(partition))
       }
 
-    def storeFetch =
-      ref.update(_.withFetch(partition, deferred))
+    def storeFetch = {
+      // revoke any previous fetch on the same (partition,streamId)
+      // TODO log a warning as this shouldn't happen!
```

Comment:
Initial logging support is now in `master` as part of #113.

Reply:
Good! I'll finish the PR tomorrow (I'm off today).

Reply:
I found another case where we send duplicate requests (downstream finishes), so I pushed a fix for that, along with rebase on master. The tests now pass without any warning log from here. Would be great if you could have a look through the pull request tomorrow. 👍